src/chainstate/stacks/db/accounts.rs

```diff
@@ -0,0 +1,829 @@
+
+            let mut shared_fees_total : u128 = 0;
+            let mut fees_total : u128 = 0;
+
+            for i in 0..sample.len() {
+                let block_miner = &sample[i].0;
+                if block_miner.fill < (fill_cutoff as u64) {
+                    // block was underfull but relayed.
+                    // The fees are entirely shared -- this will be metered at the minimum STX/compute ratio
+                    shared_fees_total = shared_fees_total.checked_add(block_miner.tx_fees_anchored << 64).expect("FATAL: STX shared-total anchored fee overflow");
+                }
+                else {
+                    // block met or exceeded budget.
+                    // The first F% is shared.
+                    // The remaining (1 - F%) is given exclusively to this miner.
+                    let shared_fees = block_miner.tx_fees_anchored.checked_mul(fill_cutoff).expect("FATAL: STX shared-total anchored fee calculation overflow");
+                    shared_fees_total = shared_fees_total.checked_add(shared_fees).expect("FATAL: STX shared-total anchored fee calculation overflow");
+
+                    if block_miner.address == miner.address {
+                        assert!((block_miner.tx_fees_anchored << 64) >= shared_fees);
+                        let exclusive_fees = (block_miner.tx_fees_anchored << 64) - shared_fees;
+                        
+                        test_debug!("{}: Anchored tx fees exclusive: {} = {} + {}", miner.address.to_string(), (fees_total + exclusive_fees) >> 64, fees_total >> 64, exclusive_fees >> 64);
+                        fees_total = fees_total.checked_add((block_miner.tx_fees_anchored << 64) - shared_fees).expect("FATAL: STX exclusive total anchored fee calculation overflow");
+                    }
+                }
+            }
+
+            let decimal_mask = 0x0000000000000000ffffffffffffffffu128;
+            let decimal_half = 0x00000000000000007fffffffffffffffu128;
+
+            anchored_tx_fee_shared_total = 
+                if shared_fees_total & decimal_mask > decimal_half {
+                    // round up
+                    (shared_fees_total >> 64) + 1
+                }
+                else {
+                    // round down
+                    shared_fees_total >> 64
+                };
+
+            anchored_tx_fee_exclusive = 
+                if fees_total & decimal_mask > decimal_half {
+                    // round up
+                    (fees_total >> 64) + 1
+                }
+                else {
+                    // round down
+                    fees_total >> 64
+                };
+            
+            test_debug!("{}: Anchored tx fees shared = {} * ({}/{})", miner.address.to_string(), anchored_tx_fee_shared_total, num_mined, sample.len());
+            anchored_tx_fee_shared = anchored_tx_fee_shared_total.checked_mul(num_mined).expect("FATAL: STX shared anchored tx fee overflow") / (sample.len() as u128);
```

Comment:
I'm not sure if this is the right ratio here -- the shared fees (according to SIP-1) should be _T_ * _P_b_ where _P_b_ is the ratio of work done by this miner over the total work of miners in that window (and _T_ is the shared reward). This code calculates _T_ * (blocks_participated_in / number_of_blocks), which is a different ratio, and also problematic because user support burns, would allow many miners to participate in a single block, meaning that the split of shares wouldn't necessarily sum to 1, e.g., in a window of 2 blocks:

block 1 = mined by Alice, supported by Bob
block 2 = mined by Bob, supported by Alice

Both Alice and Bob would have num_blocks = 2, and thus each would claim 100% share of the total shared tx fee.

Reply:
Oops -- missed your previous comment.